### ISSUE TITLE
expose BackOffOptions

### ISSUE THREAD

**Issue by eberkut** [lasith-kg/dispatch-workflow#98]
This action is really great, thanks for making it. Leveraging [exponential-backoff](https://www.npmjs.com/package/exponential-backoff) was an excellent idea but it has several default values which may not be fit for all usage.

For instance, in GHES, the default 100ms startingDelay may be to short for the API to actually return a running job even though it will eventually start without any issue.

Also, for long-running jobs, it might be worth making numOfAttempts configurable. With the default values, maximum number of attempts would be reached in 300 ms (for workflowId) or 1.5 seconds (for workflowRuns).

**Comment by lasith-kg** [comment]
Hey @eberkut! Thank you for the interest in this project! So it looks like the `exponential-backoff` library implements a multiplier on the initial `startingDelay`. This multiplier is set through the `timeMultiple` value which defaults to `2`.

This means that your corrected figures would be on the line of 700ms (for workflow ID) and 3.1 seconds (for workflow Runs). These figures are based off a starting delay of `100ms` + `2x` multiplier over `3` and `5` attempts respectively. Feel free to correct me if you think I have the incorrect interpretation. 

<img width="708" alt="image" src="https://github.com/lasith-kg/dispatch-workflow/assets/53972709/cb938106-9baa-4c02-93e1-fec3c614c41c">

From experience, the [API endpoint ](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#list-repository-workflows) that converts a workflow name (e.g `.github/workflows/blank.yaml`) into a workflow ID (`161335`) has occasionally failed on me. So the reasoning behind `exponential-backoff` is to re-attempt rather than failing on the first attempt.

**Question:**
Do you believe that we should get both `workflowID` and `workflowRuns` to share the same `exponential-backoff` parameters so that they can be centrally managed through GitHub Action default values: `starting-delay`, `number-of-attempts` and `time-multiple` 

I think this would simplify the logic within our code and the interface of our GitHub Action

**Comment by eberkut** [comment]
My calculations may be off, there were rough back of the envelope ones but at least the order of magnitude feels right and that was the main point.

Most people should not need to tweak these options and the people who do should be happy enough to have them exposed undifferentiated between workflowID and worfklowRuns so I agree that to keep the code and configuration simple they can be conflated into a single set of new parameters.

Thanks for the quick follow-up!

**Comment by lasith-kg** [comment]
I'll make an attempt to get this in the next minor release. 

I'll give you a heads up when that comes!

**Comment by lasith-kg** [comment]
Hey @eberkut! I have implemented three new parameters that can be used to tune the exponential backoff behaviour: `starting-delay`, `max-attempts` and `time-multiple`

Functionality and both unit testing and integration testing is passing 🟢! Before I merge, I will work on updating the README to reflect these advanced parameters.

```
- uses: ./
   id: dispatch-custom-backoff
   with:
     ...
     starting-delay: 150
     max-attempts: 3
     time-multiple: 1.5
```

**Comment by lasith-kg** [comment]
With release [1.7.0](https://github.com/lasith-kg/dispatch-workflow/releases/tag/v1.7.0) we provide users the ability to tune the `exponental-backoff` parameters. View here for [documentation](https://github.com/lasith-kg/dispatch-workflow#advanced-usage) on how to use these parameters. Thank you for your help @eberkut!

I'm going to leave this GitHub Issue alive for a few more days to just confirm that there are no side effects of this change. I used this time to make some quality-of-life improvements, so I made more changes than I anticipated. I would just like to make sure everything is okay

If you feel this is a cool GitHub Action, feel free to give it a ⭐! I think it would go a long way with highlighting the cool and unique functionality that is here
